addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/AVMFritzBaseBridgeHandler.java

```diff
@@ -281,16 +293,40 @@ protected void updateThingFromDevice(Thing thing, AVMFritzBaseModel device) {
     /**
      * Updates thing channels.
      *
-     * @param thing Thing which channels should be updated.
+     * @param thing     Thing which channels should be updated.
      * @param channelId ID of the channel to be updated.
-     * @param state State to be set.
+     * @param state     State to be set.
      */
     private void updateThingChannelState(Thing thing, String channelId, State state) {
-        final Channel channel = thing.getChannel(channelId);
+        Channel channel = thing.getChannel(channelId);
         if (channel != null) {
             updateState(channel.getUID(), state);
         } else {
-            logger.warn("Channel {} in thing {} does not exist, please recreate the thing", channelId, thing.getUID());
+            logger.warn("Channel '{}' in thing '{}' does not exist, recreating thing.", channelId, thing.getUID());
+            AVMFritzBaseThingHandler handler = (AVMFritzBaseThingHandler) thing.getHandler();
+            if (handler != null) {
+                handler.createChannel(channelId);
+            }
+        }
+    }
+
+    /**
+     * Updates thing channel configurations.
+     *
+     * @param thing     Thing which channel configuration should be updated.
+     * @param channelId ID of the channel which configuration to be updated.
+     * @param configId  ID of the configuration to be updated.
+     * @param value     Value to be set.
+     */
+    private void updateThingChannelConfiguration(Thing thing, String channelId, String configId, Object value) {
+        Channel channel = thing.getChannel(channelId);
+        if (channel != null) {
+            Configuration config = channel.getConfiguration();
+            // editConfiguration()
```

Comment:
What do these comments do here?

Reply:
Oh, they are not necessary. More or less a reminder for me to open an issue in ESH to ask for streamlining the API.

If we compare the `Thing` interface to the `Channel` there are some differences. For example a `Thing` has a `getProperties()` method as well as two setters `setProperty()` and `setProperties()` for them. We do not have such setters for a `Channel`. Additionally the `getProperties()` method of a `Thing` returns an immutable copy of the properties whereas the `Channel`s `getProperties()` method returns properties directly.

I will remove the comments and file the issue.